package.json

```diff
@@ -6,23 +6,30 @@
     "preinstall": "node -e \"if(process.env.CI == 'true') {console.log('Skipping preinstall...'); process.exit(1)}\" || npx -y only-allow pnpm",
     "install:csb": "pnpm install --frozen-lockfile",
     "test": "pnpm run test:ci",
-    "test:ci": "pnpm run test:format && pnpm run test:eslint && pnpm run test:lib && pnpm run test:types",
-    "test:react:17": "pnpm --filter \"./packages/react-*\" run test:lib",
-    "test:eslint": "eslint ./scripts && pnpm --filter \"./packages/**\" run test:eslint",
+    "test:ci": "nx run-many --targets=test:lib,test:types,test:eslint,test:format --parallel=5",
+    "test:react:17": "nx run-many --target=test:lib --projects=@tanstack/react-*",
+    "test:eslint": "nx run-many --target=test:eslint --parallel=5",
     "test:format": "pnpm run prettier --check",
     "test:lib": "nx run-many --target=test:lib --parallel=5",
     "test:lib:dev": "pnpm --filter \"./packages/**\" run test:lib:dev",
-    "test:build": "pnpm run build && bundlewatch && pnpm run validatePackages",
-    "test:types": "pnpm --filter \"./packages/**\" run test:types",
-    "build": "rollup --config rollup.config.js && pnpm --filter \"./packages/**\" run build && pnpm run build:types",
-    "build:types": "pnpm --filter \"./packages/**\" run build:types",
+    "test:build": "nx test:build root",
+    "test:types": "nx run-many --target=test:types --parallel=5",
+    "build": "nx run-many --target=build --projects=root,@tanstack/eslint-plugin-query",
+    "build:types": "nx run-many --target=build:types --parallel=5",
     "watch": "concurrently --kill-others \"rollup --config rollup.config.js -w\" \"pnpm run build:types --watch\"",
```

Comment:
We might be able to use `nx watch` here for the watch command - but I'll keep that separate from this pr